src/Browscap/Data/DataCollection.php

```diff
@@ -814,111 +452,12 @@ public function getDevice($device)
     }
 
     /**
-     * Get the version string identifier
-     *
-     * @return string
-     */
-    public function getVersion()
-    {
-        return $this->version;
-    }
-
-    /**
      * Get the generation DateTime object
      *
      * @return \DateTime
      */
-    public function getGenerationDate()
+    public function getGenerationDate() : \DateTime
```

Comment:
If this can't be `\DateTimeImmutable` this should `return clone $this->generationDate` to avoid action-at-a-distance